base_edifact/models/edifact.py

```diff
@@ -242,3 +242,209 @@ def map2odoo_description(self, seg):
             description = seg[2][3]
             return description
         return None
+
+    def create_segment(name, *elements):
+        cleaned_elements = []
+        replace_empty = lambda x: "" if x is False else str(x)
+        for element in elements:
+            if isinstance(element, list):
+                cleaned_elements.append([replace_empty(value) for value in element])
+            else:
+                cleaned_elements.append(replace_empty(element))
+        return Segment(*cleaned_elements)
+
+    def edifact_export_invoice(self, invoice):
+        lines = []
+        interchange = self.edifact_export_interchange(invoice)
+
+        header = self.edifact_export_header(invoice)
+        product = self.edifact_export_product(invoice)
+        summary = self.edifact_export_summary(invoice)
+        lines += header + product + summary
+        for segment in lines:
+            interchange.add_segment(self.create_segment(*segment))
+        return interchange.serialize()
+
+    def edifact_export_interchange(self, invoice):
```

Comment:
this ones looks generic enough to stay here, with a partner_id as argument instead of invoice_id